From 3039aec58203513f29edb03f84471ea941a0c226 Mon Sep 17 00:00:00 2001
From: Nicolas Chauvet <kwizart@gmail.com>
Date: Mon, 24 Sep 2018 18:28:26 +0200
Subject: [PATCH] Use @SYSTEM wide ciphers for gnutls

Gnutls upstream has support for system defined ciphers list
This is decribed at
https://fedoraproject.org/wiki/Packaging:CryptoPolicies
Also found on the debian wiki
https://wiki.debian.org/CryptoPolicy

Signed-off-by: Nicolas Chauvet <kwizart@gmail.com>
---
 modules/misc/gnutls.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/modules/misc/gnutls.c b/modules/misc/gnutls.c
index 7b63cc34c4..3ca665f3d3 100644
--- a/modules/misc/gnutls.c
+++ b/modules/misc/gnutls.c
@@ -768,12 +768,14 @@ static void CloseServer (vlc_tls_creds_t *crd)
     "hash functions and compression methods can be selected. " \
     "Refer to GNU TLS documentation for detailed syntax.")
 static const char *const priorities_values[] = {
+    "@SYSTEM",
     "PERFORMANCE",
     "NORMAL",
     "SECURE128",
     "SECURE256",
 };
 static const char *const priorities_text[] = {
+    N_("System (default to system crypto ciphers policy)"),
     N_("Performance (prioritize faster ciphers)"),
     N_("Normal"),
     N_("Secure 128-bits (exclude 256-bits ciphers)"),
-- 
2.25.4

